src/lib.rs

```diff
@@ -215,42 +232,57 @@ where
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
         let mut read: u64 = 0;
         let mut buf_left = u64::try_from(buf.len()).expect("R buflen to u64");
-        let readlimit = if let (Some(limit), _) = self.stream_cap_limit() {
-            limit
-        } else {
-            return self.stream.read(buf);
+        let Some(opts) = self.read_opt.as_ref() else {
+            return self.read_instant(buf);
         };
-        let LimiterOptions { window_time, .. } = self.read_opt.as_ref().unwrap();
         while buf_left > 0 {
             let nb_bytes_readable = self.tokens_available().0.unwrap().min(buf_left);
-            if nb_bytes_readable < readlimit.min(buf_left) {
-                let elapsed = if let Some(lrc) = self.last_read_check {
-                    lrc.elapsed()
-                } else {
-                    Duration::ZERO
-                };
-                std::thread::sleep(window_time.saturating_sub(elapsed));
-                debug_assert!(self.tokens_available().0.unwrap() > 0);
+            if nb_bytes_readable < opts.sleep_threshold.min(buf_left) {
+                let nb_left: u32 = opts
+                    .sleep_threshold
```

Comment:
Same value as the one used in comparaison maybe we could calculate it only one time